lib/connection.js

```diff
@@ -400,6 +374,7 @@ class Connection extends EventEmitter {
   // Closed socket event handler
   //
   _onSocketClose() {
+    this.closed = true;
```

Comment:
Looks like there is a typo here, it should be `this._closed` instead of `this.closed`.